desc/grid.py

```diff
@@ -733,10 +733,12 @@ def _create_nodes(  # noqa: C901
         self._endpoint = (
             t.size > 0
             and z.size > 0
-            and (t[0] == 0 and t[-1] == THETA_ENDPOINT)
-            and (z[0] == 0 and z[-1] == ZETA_ENDPOINT)
-        )
-
+            and (
```

Comment:
@unalmis I think this change is fine, what I am fixing here is if we do `LinearGrid(zeta=np.linspace(0,2*np.pi,100,endpoint=True), endpoint=True)` , this line would revert the endpoint bool to `False` because the theta grid is only `np.array([0.]`. I think that we should be able to have endpoint true for LinearGrids with a single point in one angle and then the full array in the other, and I use that in other part of the code to check if to complete a zeta curve I need to add another point, or if the grid takes care of it

Reply:
Yes this change is fine.

Since it is somewhat related, I want to remind just to make sure you're aware: we prioritize getting the surface area spacing correct over the line lengths when there are duplicate nodes. So while the correct "length" of  the zeta curve on this grid (`spacing[:, 2].sum()`) would be: $2 \pi$, it will be computed as $6.3$ instead. This number will converge to $2\pi$ as the number of points on the grid grows. 

I'm reading through the PR to see if this might mess anything up.

Reply:
Hm, I think that is fine? when we do curve computations we dont call grid.spacing

Reply:
I agree I didn't see any issues with this.